includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -294,11 +294,16 @@ public function get_intent( $intent_id ) {
 	 * @throws API_Exception - Exception thrown on setup intent creation failure.
 	 */
 	public function create_and_confirm_setup_intent( $payment_method_id, $customer_id ) {
+		$payment_method_types = [ Payment_Method::CARD ];
+		if ( WC_Payments_Features::is_sepa_enabled() ) {
+			$payment_method_types[] = Payment_Method::SEPA;
+		}
+
 		$request = [
 			'payment_method'       => $payment_method_id,
 			'customer'             => $customer_id,
 			'confirm'              => 'true',
-			'payment_method_types' => [ Payment_Method::CARD, Payment_Method::SEPA ],
+			'payment_method_types' => $payment_method_types,
```

Comment:
I think what you have here will work great. Given how strange this issue is though, I'm wondering if reverting closer to what we had before would be safer?

Previously we didn't even provide the `payment_method_types` argument (it defaults to `card` though). You can see the commit that added it here https://github.com/Automattic/woocommerce-payments/commit/a26168eff9a56f058ff08d2a40c6c23803df59cf#diff-0303646750b1a3d3223b823c3bc7345674ea5240a947c6ba00241ed4e182c41aR296.

That would also make us more consistent with `create_and_confirm_intent` which only adds the `payment_method_types` argument if the feature flag is enabled ([src](https://github.com/Automattic/woocommerce-payments/blob/85ad99cf6ac477dea8612b69a9dd220cb3b2fc5a/includes/wc-payment-api/class-wc-payments-api-client.php#L187)).

What do you think?

Reply:
Sounds good! That's better.